napari/utils/transforms/_tests/test_transform_utils.py

```diff
@@ -35,3 +44,30 @@ def test_shear_matrix_from_angle():
     matrix = shear_matrix_from_angle(35)
     np.testing.assert_almost_equal(np.diag(matrix), [1] * 3)
     np.testing.assert_almost_equal(matrix[-1, 0], np.tan(np.deg2rad(55)))
+
+
+upper = np.array([[1, 1], [0, 1]])
+lower = np.array([[1, 0], [1, 1]])
+full = np.array([[1, 1], [1, 1]])
+
+
+def test_is_upper_triangular():
+    """Test if a matrix is upper triangular."""
+    assert is_upper_triangular(upper)
+    assert not is_upper_triangular(lower)
+    assert not is_upper_triangular(full)
+
+
+def test_is_lower_triangular():
+    """Test if a matrix is lower triangular."""
+    assert not is_lower_triangular(upper)
+    assert is_lower_triangular(lower)
+    assert not is_lower_triangular(full)
+
+
+def test_check_shear_triangular():
+    """Determine shear triangular of matrix."""
+    assert check_shear_triangular(upper)
+    assert not check_shear_triangular(lower)
```

Comment:
I dislike this `assert not`, it's very unintuitive. I've left a comment further down discussion this in the `check_shear_triangular` function.